Code/Framework/AzCore/Tests/Math/Vector4Tests.cpp

```diff
@@ -18,11 +23,16 @@ namespace UnitTest
     TEST(MATH_Vector4, TestConstructors)
     {
         Vector4 v1(0.0f);
-        AZ_TEST_ASSERT((v1.GetX() == 0.0f) && (v1.GetY() == 0.0f) && (v1.GetZ() == 0.0f) && (v1.GetW() == 0.0f));
+        EXPECT_TRUE((v1.GetX() == 0.0f) && (v1.GetY() == 0.0f) && (v1.GetZ() == 0.0f) && (v1.GetW() == 0.0f));
         Vector4 v2(5.0f);
-        AZ_TEST_ASSERT((v2.GetX() == 5.0f) && (v2.GetY() == 5.0f) && (v2.GetZ() == 5.0f) && (v2.GetW() == 5.0f));
+        EXPECT_TRUE((v2.GetX() == 5.0f) && (v2.GetY() == 5.0f) && (v2.GetZ() == 5.0f) && (v2.GetW() == 5.0f));
         Vector4 v3(1.0f, 2.0f, 3.0f, 4.0f);
-        AZ_TEST_ASSERT((v3.GetX() == 1.0f) && (v3.GetY() == 2.0f) && (v3.GetZ() == 3.0f) && (v3.GetW() == 4.0f));
+        EXPECT_TRUE((v3.GetX() == 1.0f) && (v3.GetY() == 2.0f) && (v3.GetZ() == 3.0f) && (v3.GetW() == 4.0f));
```

Comment:
This should be split into 4 EXPECT_EQ messages.
As EXPECT_TRUE cannot resolve the expression statement if a failure occurs.

For example if this statement
``` EXPECT_TRUE((v1.GetX() == 0.0f) && (v1.GetY() == 0.0f) && (v1.GetZ() == 0.0f) && (v1.GetW() == 0.0f));
```
fails the error message is something like the following.

"""Expected: True
Actual: \"(v1.GetX() == 0.0f) && (v1.GetY() == 0.0f) && (v1.GetZ() == 0.0f) && (v1.GetW() == 0.0f)\" is false
"""

While using EXPECT_EQ allows each of the parameters to the macro to be resolved resulting in a better error message.
```
EXPECT_EQ(0.0f, v1.GetX());
```
"""
Expected: 0.0f
Actual: 0.01f
"""

Reply:
I'll fix these cases for this test but I can split this into another PR that fixes the rest of these cases in the tests.

Reply:
You don't have to to fix up the existing test cases since those are unrelated to your changes.
But if you want to do it as part of another PR, go ahead!